presto-main/src/main/java/com/facebook/presto/operator/scalar/StringFunctions.java

```diff
@@ -721,7 +721,7 @@ public static long levenshteinDistance(@SqlType("varchar(x)") Slice left, @SqlTy
     @SqlType(StandardTypes.BIGINT)
     public static long hammingDistance(@SqlType("varchar(x)") Slice left, @SqlType("varchar(y)") Slice right)
     {
-        int distance = 0;
+        long distance = 0;
```

Comment:
@satybald Are you suggesting that we limit the maximum length of the string the function should take?